class/track/line/SolidLine.js

```diff
@@ -12,20 +12,27 @@ export class SolidLine extends Line {
             return this;
         }
         this.touched = true;
-        let pos = object.pos,
-            vel = object.velocity,
-            radius = object.size,
-            diff, // The vector from the closest point on the line to the object
-            dist,
-            Ap = pos.sub(this.a),
-            u = Ap.dot(this.vector) / this.len / this.len;  // Named "u" because it is sort of like uv coordinates
+        let pos = object.pos;
+        let vel = object.velocity;
+        let radius = object.size;
+        let pp = this.vector;
+        let diff; // The vector from the closest point on the line to the object
+        let dist;
+        let aDiff = pos.sub(this.a);
+        let u = aDiff.dot(this.vector) / this.len / this.len;  // Named "u" because it is sort of like uv coordinate
         if (u >= 0 && u <= 1) {
-            // "sign" is negative if the center of the mass has passed through the line.
-            let sign = (Ap.x * this.vector.y - Ap.y * this.vector.x) * ((Ap.x - vel.x) * this.vector.y - (Ap.y - vel.y) * this.vector.x) < 0 ? -1 : 1;
-            diff = Ap.sub(this.vector.scale(u));
+            // "passedThrough" is negative if the center of the mass has passed through the line.
+            let passedThrough = (aDiff.x * this.vector.y - aDiff.y * this.vector.x) * ((aDiff.x - vel.x) * this.vector.y - (aDiff.y - vel.y) * this.vector.x) < 0 ? -1 : 1;
+            if (passedThrough === -1) {
+                let measure = (aDiff.y * vel.x - aDiff.x * vel.y) / (pp.y * vel.x - pp.x * vel.y);
+                if (measure < 0 || measure > 1) {
+                    return;
```

Comment:
Shouldn't it be `return this;` ? Like the one below

Reply:
Good catch, I'll change it.